src/player.js

```diff
@@ -1,23 +1,32 @@
 import VideoPlayer from './video-player';
 import { defaultProfiles } from './config/profiles';
 import { isRawUrl } from './plugins/cloudinary/common';
+import { unsigned_url_prefix } from '@cloudinary/url-gen/backwards/utils/unsigned_url_prefix';
 
-export const getProfile = async (cloudName, profile) => {
+export const getProfile = async (profile, initOptions) => {
   if (Object.keys(defaultProfiles).includes(profile)) {
     return defaultProfiles[profile];
   }
 
-  if (isRawUrl(profile)) {
-    return await fetch(profile, { method: 'GET' }).then(res => res.json());
-  }
+  const urlPrefix = unsigned_url_prefix(
+    null,
+    initOptions.cloudName || initOptions.cloud_name,
+    initOptions.private_cdn,
+    initOptions.cdn_subdomain,
+    initOptions.secure_cdn_subdomain,
+    initOptions.cname,
+    initOptions.secure,
+    initOptions.secure_distribution,
+  );
 
-  throw new Error('Custom profiles will be supported soon, please use one of default profiles: "cldDefault", "cldLooping" or "cldAdaptiveStream"');
+  const profileUrl = isRawUrl(profile) ? profile : `${urlPrefix}/_applet_/video_service/video_player_profiles/${profile}.json`;
+  return await fetch(profileUrl, { method: 'GET' }).then(res => res.json());
```

Comment:
I think the await here and the async in the function declaration are both redundant, but especially here it is really not needed
```suggestion
  return fetch(profileUrl, { method: 'GET' }).then(res => res.json());
```

Reply:
changed, I would like to keep entire method as async just in case - so w need to have async declaration on the top but there await can be removed